scp/cc/azure/attestation/src/attestation.h

```diff
@@ -0,0 +1,149 @@
+                                  // author key is present in
+                                  // AUTHOR_KEY_DIGEST. Set to the value of
+                                  // GCTX.AuthorKeyEn.
+  uint32_t reserved1;             // must be zero
+  uint8_t report_data[64];        // Guest provided data.
+  uint8_t measurement[48];        // measurement calculated at launch
+  uint8_t host_data[32];          // data provided by the hypervisor at launch
+  uint8_t id_key_digest[48];      // SHA-384 digest of the ID public key that
+                                  // signed the ID block provided in
+                                  // SNP_LAUNCH_FINISH
+  uint8_t author_key_digest[48];  // SHA-384 digest of the Author public key
+                                  // that certified the ID key, if provided in
+                                  // SNP_LAUNCH_FINISH. Zeros if author_key_en
+                                  // is 1 (sounds backwards to me).
+  uint8_t report_id[32];          // Report ID of this guest.
+  uint8_t report_id_ma[32];       // Report ID of this guest's mmigration
+                                  // agent.
+  uint64_t reported_tcb;          // Reported TCB version used to derive the
+                                  // VCEK that signed this report
+  uint8_t reserved2[24];          // reserved
+  uint8_t chip_id[64];            // Identifier unique to the chip
+  uint8_t committed_svn[8];       // The current commited SVN of the firware
+                                  // (version 2 report feature)
+  uint8_t committed_version[8];   // The current commited version of the
+                                  // firware
+  uint8_t launch_svn[8];          // The SVN that this guest was launched or
+                                  // migrated at
+  uint8_t reserved3[168];         // reserved
+  uint8_t signature[512];         // Signature of this attestation report.
+                                  // See table 23.
+};
+
+/* from SEV-SNP Firmware ABI Specification Table 22 */
+struct SnpResponse {
+  uint32_t status;
+  uint32_t report_size;
+  uint8_t reserved[24];
+  SnpReport report;
+  uint8_t padding[64];  // padding to the size of SEV_SNP_REPORT_RSP_BUF_SZ
+                        // (i.e., 1280 bytes)
+};
+
+enum SnpType { SEV, SEV_GUEST, NONE };
+
+struct AttestationReport {
+  std::string evidence;
+  std::string endorsements;
+  std::string uvm_endorsements;
+  std::string endorsed_tcb;
+
+  operator nlohmann::json() const {
+    return nlohmann::json{{"evidence", evidence},
+                          {"endorsements", endorsements},
+                          {"uvm_endorsements", uvm_endorsements},
+                          {"endorsed_tcb", endorsed_tcb}};
+  }
+};
+
+SnpType getSnpType();
+
+bool hasSnp();
+
+AttestationReport fetchSnpAttestation(const std::string report_data = "");
+
+AttestationReport fetchFakeSnpAttestation();
+
+std::string getSnpEvidence(const std::string report_data);
```

Comment:
I think it would be safer to keep it as a string for now, we might eventually want to let the user provide a string report data when calling `:print_report`

Reply:
I agree that we can change when it's actually necessary